ui/pages/confirmations/components/confirm/footer/useIsDangerButton.ts

```diff
@@ -0,0 +1,29 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import useSignatureSecurityAlertResponse from '../../../hooks/useSignatureSecurityAlertResponse';
+import { currentConfirmationSelector } from '../../../selectors';
+import { SecurityAlertResponse } from '../../../types/confirm';
+import { BlockaidResultType } from '../../../../../../shared/constants/security-provider';
+
+const useIsDangerButton = () => {
+  const [isDangerButton, setIsDangerButton] = useState(false);
+  const currentConfirmation = useSelector(currentConfirmationSelector);
+
+  const currentSecurityAlertId = (
+    currentConfirmation?.securityAlertResponse as SecurityAlertResponse
+  )?.securityAlertId;
+
+  const signatureSecurityAlertResponse = useSignatureSecurityAlertResponse(
+    currentSecurityAlertId,
+  );
+
+  useEffect(() => {
```

Comment:
Why are we using a `useEffect` here and storing this value in a `useState`?

We have no asynchronous logic to run so the `useEffect` shouldn't be needed.

We could use a `useMemo` if the logic was complex and time consuming but this has almost no performance cost.

Would it be simpler to remove both the `useEffect` and `useState` and just return the comparison directly such as:

```
return signatureSecurityAlertResponse?.result_type === BlockaidResultType.Malicious;
```

Reply:
you're right. updated [1fae5a8](https://github.com/MetaMask/metamask-extension/pull/24402/commits/1fae5a805b7e1f3cb916d64de71dd318e78c0064)